frontend/src/pages/projects/screens/detail/storage/ClusterStorageModal.tsx

```diff
@@ -1,161 +1,359 @@
     >
-      {workbenchEnabled && (
-        <>
-          {hasExistingNotebookConnections && (
-            <ExistingConnectedNotebooks
-              connectedNotebooks={removableNotebooks}
-              onNotebookRemove={(notebook: NotebookKind) =>
-                setRemovedNotebooks([...removedNotebooks, notebook.metadata.name])
-              }
-              loaded={removableNotebookLoaded}
-              error={removableNotebookError}
-            />
-          )}
-          <StorageNotebookConnections
-            setForNotebookData={(forNotebookData) => {
-              setNotebookData(forNotebookData);
-            }}
-            forNotebookData={notebookData}
-            connectedNotebooks={connectedNotebooks}
-          />
-          {restartNotebooks.length !== 0 && (
-            <FormGroup>
-              <NotebookRestartAlert notebooks={restartNotebooks} />
-            </FormGroup>
-          )}
-        </>
-      )}
+      <>
+        {workbenchEnabled && (
+          <FormGroup label="Workbench connections" fieldId="workbench-connections">
+            <Stack hasGutter>
+              <StackItem>
+                <Table
+                  data-testid="workbench-connection-table"
+                  variant="compact"
+                  columns={storageColumns}
+                  data={notebookData}
+                  rowRenderer={(row, rowIndex) => (
+                    <ClusterStorageTableRow
+                      key={rowIndex}
+                      obj={row}
+                      inUseMountPaths={getInUseMountPaths(
+                        row.name,
+                        allNotebooks,
+                        existingPvc?.metadata.name,
+                      )}
+                      availableNotebooks={{
+                        notebooks: [
+                          ...availableNotebooks,
+                          ...allNotebooks.filter(
+                            (currentData) => currentData.metadata.name === row.name,
+                          ),
+                        ],
+                        loaded,
+                      }}
+                      onMountPathUpdate={(value, format) =>
+                        handleMountPathUpdate(rowIndex, value, format)
+                      }
+                      onNotebookSelect={(notebookName) => {
+                        const updatedData = [...notebookData];
+                        updatedData[rowIndex] = {
+                          ...row,
+                          name: notebookName,
+                          mountPath: {
+                            ...row.mountPath,
+                            error: validateClusterMountPath(
+                              row.mountPath.value,
+                              getInUseMountPaths(
+                                notebookName,
+                                allNotebooks,
+                                existingPvc?.metadata.name,
+                              ),
```

Comment:
@Gkrumbach07 I made this changes from your fix to a handle edge cases

Reply:
yeah np, i didnt look too much into the specifics when i posted it